python/pyarrow/includes/libarrow.pxd

```diff
@@ -2934,3 +2966,10 @@ cdef extern from "arrow/util/byte_size.h" namespace "arrow::util" nogil:
     int64_t TotalBufferSize(const CChunkedArray& array)
     int64_t TotalBufferSize(const CRecordBatch& record_batch)
     int64_t TotalBufferSize(const CTable& table)
+
+cdef extern from "arrow/python/udf.h" namespace "arrow::py" nogil:
```

Comment:
This shouldn't be "nogil" as `RegisterScalarFunction` takes a Python object.

Reply:
Good catch.